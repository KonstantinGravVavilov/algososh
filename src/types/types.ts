import { ElementStates } from "./element-states";

export type TString = {
  letter: string;
  state: ElementStates;
};

export type TArray = {
  val: number;
  state: ElementStates;
};

export type TQueue = {
  val: string | null;
  state: ElementStates;
};

export type TListElem = {
  val: string;
  upCircle: React.ReactElement | null;
  downCircle: React.ReactElement | null;
  next: boolean;
};
