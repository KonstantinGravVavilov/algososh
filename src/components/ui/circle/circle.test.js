import renderer from "react-test-renderer";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe('testing Circle component', () => {
  it('no letter in circle', () => {
    const circle = renderer
      .create(<Circle letter="" />)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('letters in circle', () => {
    const circle = renderer
      .create(<Circle letter="lttr" />)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('head in circle', () => {
    const circle = renderer
      .create(<Circle head="hd" />)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('react-elem in head in circle', () => {
    const circle = renderer
      .create(<Circle head={ <Circle /> } />)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('tail in circle', () => {
    const circle = renderer
      .create(<Circle tail="hd" />)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('react-elem in tail in circle', () => {
    const circle = renderer
      .create(<Circle tail={ <Circle /> } />)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('index in circle', () => {
    const circle = renderer
      .create(<Circle index="indx" />)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('props isSmall in circle', () => {
    const circle = renderer
      .create(<Circle isSmall={true} />)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('default circle', () => {
    const circle = renderer
      .create(<Circle state={ ElementStates.Default } />)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('changing circle', () => {
    const circle = renderer
      .create(<Circle state={ ElementStates.Changing } />)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('modified circle', () => {
    const circle = renderer
      .create(<Circle state={ ElementStates.Modified } />)
      .toJSON();
    expect(circle).toMatchSnapshot();
  });

});
