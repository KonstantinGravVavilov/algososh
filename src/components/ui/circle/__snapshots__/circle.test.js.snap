// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Circle component changing circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`testing Circle component default circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`testing Circle component head in circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    hd
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`testing Circle component index in circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    indx
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`testing Circle component letters in circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      lttr
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`testing Circle component modified circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`testing Circle component no letter in circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`testing Circle component props isSmall in circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`testing Circle component react-elem in head in circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`testing Circle component react-elem in tail in circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`testing Circle component tail in circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    hd
  </div>
</div>
`;
